### DISCUSSION TITLE
Why Filter web part is not showing any filters

### DISCUSSION THREAD

**Discussion by mvcsharepointdev** [microsoft-search/pnp-modern-search#3932] [Q&A]
I have 2 refinable strings linked to 2 SharePoint choice columns , as follow:-

![image](https://github.com/user-attachments/assets/eacfaf33-e44d-4c2e-b798-e08ec8075850)

the columns are added to 10 SharePoint lists, and they contain data + I submit a request to do a full crawl 3 days ago. but when i added those 2 refinable/s inside my Filter web part, nothing has been shown? 

Here are the filters:-

![image](https://github.com/user-attachments/assets/8b09d9b6-d04e-4350-a563-49eb48777ef6)

and I linked the search filters web part to the search result web part + I linked to the search result web part to the search filter web part. any advice?

![image](https://github.com/user-attachments/assets/85cb9ad9-3a5f-4d84-887e-a69f6aad3a8d)

**Comment by kasperbolarsen** [comment]
Pls see https://microsoft-search.github.io/pnp-modern-search/using-query-tools-to-verify-issues/

As step 1 should always be to check if the api returns the data you expect.  I prefer to use SP Editor over the SharePoint Search Query tool,  but that is just a question of taste.

**Comment by calebmil** [comment]
If you recently defined your managed property mappings, you may need to reindex your library(ies) to get the index updated to reflect that mapping.  I feel like this was not the case in the past but have found that I have to do this quite frequently in recent months.

**Reply by mvcsharepointdev** [reply]
@calebmil we have 10 custom libraries using the same content type, so i have to re-index all the lists? i did a reindex for one list and the whole site last week

**Reply by kasperbolarsen** [reply]
so you did use SP Editor or the Search Query tool to inspect the data returned by the API ?

**Reply by mvcsharepointdev** [reply]
@kasperbolarsen never used those tools, but what is the benefit to use those and how to use them?
Thanks

**Reply by calebmil** [reply]
Have you verified that the managed property is in the selected properties field?  Have you looked at the results web part in debug layout?  I find that my mapped property fields show up with null as a value when I expected them to have values.  That's my sign to reindex.

**Reply by mvcsharepointdev** [reply]
@calebmil @kasperbolarsen  ok i used the "SharePoint Search query toolv2.10.0", and i checked the managed properties for an item, and I got this result:-

![image](https://github.com/user-attachments/assets/7c6a80bb-b4de-43e4-ba65-050b171b61c6)


so the RefinableDate00 which is linked to a Date/Time column has a value..
while RefinableDate01 which is linked to a Calculated column of type Date/Time is empty...

here are my refinables:-

![image](https://github.com/user-attachments/assets/597ed7c1-a980-4ea9-80db-239edbda6ba3)


so the issue is within sharepoint search, that it did not crawl the calculated column yet, although i request a re-index to the list and to the site 1 week ago.... or this might be by design, that calculated column will always be empty inside search !! not sure

**Reply by kasperbolarsen** [reply]
please look at https://microsoft-search.github.io/pnp-modern-search/using-query-tools-to-verify-issues/  it contains a lot of tips on solving this kind of issues

**Reply by mvcsharepointdev** [reply]
@kasperbolarsen i used it as mentioned above

**Reply by mvcsharepointdev** [reply]
@calebmil @kasperbolarsen also i checked the crawl properties, where my calcualted column name is CalAMPM, and the crawl property is there, as follow:-

![image](https://github.com/user-attachments/assets/9ebeb5a2-34b0-48a8-ad9f-58e388fd9bee)

**Comment by kasperbolarsen** [comment]
since this is a SharePoint Search issue you can create a service ticket with MS. Please remember NOT to mention PnP Modern Search in the comms with MS, as we have see multiple cases where the service agent try to close the ticket with an excuse about it being an issue with PnP Modern search

**Reply by mvcsharepointdev** [reply]
@kasperbolarsen yes MS support always try to blame others, already opened many tickets with them, and till now i did not get any answer from their side about this issue

**Reply by kasperbolarsen** [reply]
You might get additional info from this blog post of mine about content not getting indexed,https://www.m365thinking.com/post/what-to-do-when-the-sharepoint-search-doesn-t-pick-up-your-content-part-1-of-4
